### PR TITLE
feat(run-pre-commit): Add support to install nix

### PR DESCRIPTION
**Pull request by Techassi**
This adds support to install nix when running pre-commit. This is useful for repositories which have pre-commit hooks requiring nix to run them.

### CODE REVIEW

File: run-pre-commit/action.yml [hunk 1]
```diff
@@ -50,6 +56,20 @@ runs:
 
         echo "$LOCATION_DIR" | tee -a "$GITHUB_PATH"
 
+    - name: Abort nix-github-token input is unset
+      if: ${{ inputs.nix && !inputs.nix-github-token }}
```
Comment (NickLarsenNZ) [inline run-pre-commit/action.yml:60]: FYI, the interpolation isn't required.

```suggestion
      if: inputs.nix && !inputs.nix-github-token
```

File: run-pre-commit/action.yml [hunk 2]
```diff
@@ -50,6 +56,20 @@ runs:
 
         echo "$LOCATION_DIR" | tee -a "$GITHUB_PATH"
 
+    - name: Abort nix-github-token input is unset
```
Comment (NickLarsenNZ) [inline run-pre-commit/action.yml:59]:
```suggestion
    - name: Abort if nix-github-token input is not set
```

File: run-pre-commit/action.yml [hunk 3]
```diff
@@ -50,6 +56,20 @@ runs:
...
+      if: ${{ inputs.nix && !inputs.nix-github-token }}
+      shell: bash
+      run: |
+        echo "nix-github-token input must be set when nix input is set"
+        exit 1
+
+    - name: Setup nix
+      if: ${{ inputs.nix }}
```
Comment (NickLarsenNZ) [inline run-pre-commit/action.yml:67]:
```suggestion
      if: inputs.nix
```

**Review by NickLarsenNZ — DISMISSED** [review]
LGTM, some minor suggestions.

**Comment by Techassi** [comment]
I didn't, but I will over in https://github.com/stackabletech/stackable-cockpit

**Pull request merged**